template/src/ontology/run.sh.jinja2

```diff
@@ -65,18 +65,18 @@ ROBOT_JAVA_ARGS=$ODK_JAVA_OPTS
 JAVA_OPTS=$ODK_JAVA_OPTS
 GH_TOKEN=$GH_TOKEN
 EOF
-
+{% endif %}
 if [ -n "$USE_SINGULARITY" ]; then
     {% if project.use_external_date is sameas True %}export TODAY=$(date +%Y-%m-%d){% endif %}
-    singularity exec --cleanenv $ODK_SINGULARITY_OPTIONS \
-        --env-file run.sh.env \
+    singularity exec --cleanenv $ODK_SINGULARITY_OPTIONS \{% if project.use_env_file_docker is sameas True %}
+        --env-file run.sh.env \{% endif %}
         --bind $VOLUME_BIND \
         -W $WORK_DIR \
         docker://obolibrary/$ODK_IMAGE:$ODK_TAG $TIMECMD "$@"
 else
     BIND_OPTIONS="-v $(echo $VOLUME_BIND | sed 's/,/ -v /')"
-    docker run -u $(id -u):$(id -g) $ODK_DOCKER_OPTIONS $BIND_OPTIONS -w $WORK_DIR \
-        --env-file run.sh.env \
+    docker run {% if project.set_docker_user is sameas True %}-u $(id -u):$(id -g) {% endif %}$ODK_DOCKER_OPTIONS $BIND_OPTIONS -w $WORK_DIR \{% if project.use_env_file_docker is sameas True %}
+        --env-file run.sh.env \{% endif %}
```

Comment:
So if `use_env_file_docker` is False (which is the default), _no_ environment variables are passed to the container, which means that `ROBOT_JAVA_ARGS` (which is used to set the amount of memory ROBOT can use) is not passed either. Is this really what we want?

Reply:
s**t didn't think of this.. This ODK release will never happen.

I just cant have so many of my pipelines breaking now with introducing a non-configurable option.. Literally _all_ pipelines I run have a small requirements.txt install step associated with it now. I don't know what else to do - I don't want other peoples stuff to break now the same way, especially since it is really hard to debug this.

Reply:
> Literally all pipelines I run have a small requirements.txt install step associated with it now. 

How is that related to avoiding the use of the `run.sh.env` file?

Reply:
Sorry, I read your comment wrong. This one I had to make optional because for some reason, sometimes my docker crashes / the container does not start, and removing the run.sh.env logic causes this problem not to happen -- I don't have a clue how they relate though. It makes no sense. I just guessed that maybe it has something to do with the `trap` but maybe it is entirely unrelated..